tests/async/tasyncrecursion.nim

```diff
@@ -17,5 +17,6 @@ proc asyncRecursionTest*(): Future[int] {.async.} =
     inc(i)
 
 when isMainModule:
+  setGlobalDispatcher(newDispatcher())
```

Comment:
@zielmicha, can you revert it please? The test runs just fine, creating the dispatcher lazily.